src/main/java/org/squbs/sample/SampleActor.java

```diff
@@ -0,0 +1,59 @@
+package org.squbs.sample;
+
+import akka.NotUsed;
+import akka.actor.AbstractActor;
+import akka.actor.ActorRef;
+import akka.actor.Cancellable;
+import akka.japi.pf.ReceiveBuilder;
+import akka.stream.Attributes;
+import akka.stream.DelayOverflowStrategy;
+import akka.stream.javadsl.Source;
+import scala.PartialFunction;
+import scala.runtime.BoxedUnit;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static org.squbs.sample.AppConstants.chunkEnd;
+import static org.squbs.sample.AppConstants.emptyRequest;
+
+/**
+ * This is the actor that handles the request messages.
+ */
+public class SampleActor extends AbstractActor {
+
+    public SampleActor() {
+        receive(ReceiveBuilder.
+            match(PingRequest.class, r -> {
+                if (r.who.trim().length() > 0) {
+                    sender().tell(new PingResponse("Hello " + r.who + " welcome to squbs!"), self());
+                } else {
+                    sender().tell(emptyRequest, self());
+                }
+                context().stop(self());
+            }).
+            match(ChunkRequest.class, r -> {
+                ActorRef requester = sender(); // Save the requester for use in the scheduler.
+
+                String[] responseArray = {"Hello ", r.who, " welcome ", "to ", "squbs!"};
+                Stream<String> responseStream = Arrays.stream(responseArray);
+
+                Source<PingResponse, NotUsed> source = Source.from(Arrays.asList(responseArray))
+                    .map(elem -> new PingResponse(elem))
+                    .concat(Source.single(new PingResponse(chunkEnd)))
```

Comment:
Would it be better to concat the end in the service? The business logic does not even have to care. Just send a source. The service will do what is needed by the protocol. What do you think?